fuel-vm/src/error.rs

```diff
@@ -223,6 +223,39 @@ impl From<Infallible> for io::Error {
     }
 }
 
+/// Predicates checking failed
+#[derive(Debug, Error)]
+pub enum PredicateVerificationFailed {
+    /// The transaction doesn't contain enough gas to evaluate the predicate
+    #[error("Insufficient gas available for predicates")]
+    OutOfGas,
+    /// The predicate owner does not correspond to the predicate code
+    #[error("Predicate owner invalid, doesn't match code root")]
+    InvalidOwner,
+    /// The predicate wasn't successfully evaluated to true
+    #[error("Predicate failed to evaluate")]
+    False,
+    /// An unexpected error occurred.
+    #[error(transparent)]
+    Io(#[from] io::Error),
+}
+
+impl From<PredicateVerificationFailed> for CheckError {
+    fn from(_: PredicateVerificationFailed) -> Self {
+        CheckError::PredicateVerificationFailed
```

Comment:
The error is an empty newtype. We don't currently collect any error information.

Reply:
I meant to add the `PredicateVerificationFailed` into `CheckError::PredicateVerificationFailed`, because they all leave in the `fuel-vm` now, we can do that

Reply:
I still don't understand what you mean

Reply:
Never mind, `CheckError` lives in the `fuel-tx`. We can't use it in the `PredicateVerificationFailed` from `fuel-vm` there